gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -276,39 +259,46 @@ hybridElimination(const HybridGaussianFactorGraph &factors,
   auto gaussianMixture = boost::make_shared<GaussianMixture>(
       frontalKeys, continuousSeparator, discreteSeparator, conditionals);
 
-  // If there are no more continuous parents, then we should create a
-  // DiscreteFactor here, with the error for each discrete choice.
   if (continuousSeparator.empty()) {
-    auto factorProb =
-        [&](const GaussianMixtureFactor::FactorAndConstant &factor_z) {
-          // This is the probability q(μ) at the MLE point.
-          // factor_z.factor is a factor without keys,
-          // just containing the residual.
-          return exp(-factor_z.error(VectorValues()));
-        };
-
-    const DecisionTree<Key, double> fdt(newFactors, factorProb);
-    // // Normalize the values of decision tree to be valid probabilities
-    // double sum = 0.0;
-    // auto visitor = [&](double y) { sum += y; };
-    // fdt.visit(visitor);
-    // // Check if sum is 0, and update accordingly.
-    // if (sum == 0) {
-    //   sum = 1.0;
-    // }
-    // fdt = DecisionTree<Key, double>(fdt,
-    //                                 [sum](const double &x) { return x / sum;
-    //                                 });
-    const auto discreteFactor =
-        boost::make_shared<DecisionTreeFactor>(discreteSeparator, fdt);
+    // If there are no more continuous parents, then we create a
+    // DiscreteFactor here, with the error for each discrete choice.
+
+    // Integrate the probability mass in the last continuous conditional using
+    // the unnormalized probability q(μ;m) = exp(-error(μ;m)) at the mean.
+    //   discrete_probability = exp(-error(μ;m)) * sqrt(det(2π Σ_m))
+    auto probability = [&](const Result &pair) -> double {
+      static const VectorValues kEmpty;
+      // If the factor is not null, it has no keys, just contains the residual.
+      const auto &factor = pair.second;
+      if (!factor) return 1.0;  // TODO(dellaert): not loving this.
```

Comment:
How could this branch be taken?

Reply:
Yeah, whenever we do BayesTree there is some way (that I don't yet understand) that nullptrs are used, and we are dealing with all over the place. Can you explain? Or @varunagrawal ?